homeassistant/config_entries.py

```diff
@@ -422,6 +424,11 @@ def __init__(
         _setter(self, "_tries", 0)
         _setter(self, "created_at", created_at or utcnow())
         _setter(self, "modified_at", modified_at or utcnow())
+        _setter(
+            self,
+            "discovery_keys",
+            tuple(MappingProxyType(dict(key)) for key in discovery_keys),
```

Comment:
The reason for the `MappingProxyType(dict(key))` is to avoid multiple layers of `mappingproxy`. Maybe it would be better to use a helper function which inspects each item and only wraps mappingproxy when needed?